Runtimes/Core/CMakeLists.txt

```diff
@@ -103,6 +103,13 @@ option(SwiftCore_ENABLE_FILESYSTEM_SUPPORT "Build for systems that have a filesy
 option(SwiftCore_ENABLE_OS_TRACE_LAZY_INIT "Use os_trace call to check if lazy init has been completed before making os_signpost calls." ${SwiftCore_HAS_OS_TRACE})
 option(SwiftCore_HAS_DARWIN_LIBMALLOC "Use Darwin malloc features" ${APPLE})
 
+if("${APPLE}" OR "${LINUX}")
+  set(SwiftCore_ENABLE_PRESPECIALIZATION_default ON)
+else()
+  set(SwiftCore_ENABLE_PRESPECIALIZATION_default OFF)
+endif()
+option(SwiftCore_ENABLE_PRESPECIALIZATION "Enable generic metadata prespecialization" "${SwiftCore_ENABLE_PRESPECIALIZATION_default}")
```

Comment:
```suggestion
option(SwiftCore_ENABLE_PRESPECIALIZATION "Enable generic metadata prespecialization" "$<OR:$<BOOL:${APPLE}>,$<BOOL:${LINUX}>>")
```

Reply:
This will not work according to the [Cmake documentation](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#id3)

> Generator expressions are allowed in the context of many target properties [...] They may also be used when using commands to populate those properties